openstack/blockstorage/v2/extensions/volumeactions/requests.go

```diff
@@ -0,0 +1,176 @@
+package volumeactions
+
+import (
+	"github.com/rackspace/gophercloud"
+)
+
+type AttachOptsBuilder interface {
+	ToVolumeAttachMap() (map[string]interface{}, error)
+}
+
+type AttachOpts struct {
+	// The mountpoint of this volume
+	MountPoint string
+	// The nova instance ID, can't set simultaneously with HostName
+	InstanceUUID string
+	// The hostname of baremetal host, can't set simultaneously with InstanceUUID
+	HostName string
+	// Mount mode of this volume
+	Mode string
```

Comment:
will this only ever be `ro` or `rw`, or do we expect custom values? if the former, perhaps an enum is better:

```go
// Mode describes the attachment mode for volumes.
type Mode string

// These constants determine how a volume is attached
const (
	ReadOnly Mode = "ro"
	ReadWrite Mode = "rw"
)

type AttachOpts struct {
	...
	Mode Mode
	...
}
```